types/ember/test/ember-module-tests.ts

```diff
@@ -0,0 +1,280 @@
+const a1: Ember.Array<string> = [];
+// Ember.ArrayProxy
+new Ember.ArrayProxy<number>([3, 3, 2]); // $ExpectType ArrayProxy<number>
+// Ember.Checkbox
+const cb = new Ember.Checkbox(); // $ExpectType Checkbox
+cb.tagName; // $ExpectType string
+// Ember.Component
+const C1 = Ember.Component.extend({ classNames: ['foo'] });
+class C2 extends Ember.Component {
+    classNames = ['foo'];
+}
+const c1 = new C1();
+const c2 = new C2();
+C1.create();
+C2.create();
+c1.didInsertElement();
+c2.didInsertElement();
+// Ember.ComputedProperty
+const cp: Ember.ComputedProperty<string, string> = Ember.computed('foo', {
+    get(): string {
+        return '';
+    },
+    set(_key: string, newVal: string): string {
+        return '';
+    }
+});
+// Ember.ContainerDebugAdapter
+const cda = new Ember.ContainerDebugAdapter(); // $ExpectType ContainerDebugAdapter
+// Ember.Controller
+const con = new Ember.Controller(); // $ExpectType Controller
+// Ember.CoreObject
+const co = new Ember.CoreObject(); // $ExpectType CoreObject
+// Ember.DataAdapter
+const da = new Ember.DataAdapter(); // $ExpectType DataAdapter
+// Ember.Debug
+Ember.Debug.registerDeprecationHandler(() => {});
+Ember.Debug.registerWarnHandler(() => {});
+// Ember.DefaultResolver
+const dr = new Ember.DefaultResolver();
+dr.resolve('route:index');
+// Ember.Engine
+const e1 = new Ember.Engine();
+e1.register('data:foo', {}, { instantiate: false });
+// Ember.EngineInstance
+const ei1 = new Ember.EngineInstance();
+ei1.lookup('data:foo');
+// Ember.Error
+new Ember.Error('Halp!');
+// Ember.Evented
+const oe1 = EmberObject.extend(Evented).create();
+oe1.trigger('foo');
+oe1.on('bar', () => {});
+// Ember.HashLocation
+const hl = new Ember.HashLocation();
+// Ember.Helper
+const h1 = Ember.Helper.extend({
+    compute() {
+        this.recompute();
+        return '';
+    }
+});
+// Ember.HistoryLocation
+const hil = new Ember.HistoryLocation();
+// Ember.Instrumentation
+Ember.Instrumentation.instrument('foo', {}, () => {}, window);
+Ember.Instrumentation.reset();
+// Ember.LinkComponent
+Ember.LinkComponent.create(); // $ExpectType LinkComponent
+// Ember.Mixin
+Ember.Object.extend(Ember.Mixin.create({ foo: 'bar' }), {
+    baz() {
+        this.foo; // $ExpectType string
+    }
+});
+// Ember.MutableArray
+const ma1: Ember.MutableArray<string> = [
+    'money',
+    'in',
+    'the',
+    'bananna',
+    'stand'
+];
+ma1.addObject('!');
+ma1.filterBy('');
+// Ember.MutableEnumerable
+// tslint:disable-next-line:prefer-const
+let me1: Ember.MutableEnumerable<[string]> = null as any;
+me1.compact();
+// Ember.Namespace
+const myNs = Ember.Namespace.extend({});
+// Ember.NativeArray
+const na: Ember.NativeArray<number> = Ember.A([2, 3, 4]);
+na; // $ExpectType NativeArray<number>
+na.clear();
+// Ember.NoneLocation
+new Ember.NoneLocation();
+// Ember.Object
+new Ember.Object();
+// Ember.ObjectProxy
+new Ember.ObjectProxy();
+// Ember.Observable
+Ember.Object.extend(Ember.Observable, {});
```

Comment:
check that observable methods are present

Reply:
I'm not sure that this would be useful, since `EmberObject` already includes the `Observable` mixin (the methods will be present already). This test is designed to ensure that `Ember.Observable` can be regarded as a mixin